enhancements/api-review/add-ImagePolicy-to-config.openshift.io-v1alpha1.md

```diff
@@ -0,0 +1,585 @@
+
+// ImagePolicy holds namespace-wide configuration configuration for image signature verification
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+// +openshift:enable:FeatureSets=TechPreviewNoUpgrade
+type ImagePolicy struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// metadata is the standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec holds user settable values for configuration
+	// +kubebuilder:validation:Required
+	// +required
+	Spec ImagePolicySpec `json:"spec"`
+	// status contains the observed state of the resource.
+	// +optional
+	Status ImagePolicyStatus `json:"status,omitempty"`
+}
+
+// ImagePolicySpec is the specification of the ImagePolicy CRD.
+type ImagePolicySpec struct {
+	// scopes defines the list of image identities assigned to a policy. Each item refers to a scope in a registry implementing the "Docker Registry HTTP API V2".
+	// scopes matching individual images are named Docker references in the fully expanded form, either using a tag or digest. For example, docker.io/library/busybox:latest (not busybox:latest).
+	// More general scopes are prefixes of individual-image scopes, and specify a repository (by omitting the tag or digest), a repository
+	// namespace, or a registry host (by only specifying the host name and possibly a port number) or a wildcarded expression starting with *., for matching all subdomains (not including a port number).
+	// For wildcarded subdomain matching, *.example.com is a valid case, but example*.*.com is not.
+	// For more information about the format, see the document
+	// about the docker transport field:
+	// https://github.com/containers/image/blob/main/docs/containers-policy.json.5.md#docker
+	// +kubebuilder:validation:Required
+	// +required
+	// +listType=set
+	Scopes []ImageScope `json:"scopes"`
+	// policy defines the verification policy for the items in the scopes list
+	// +kubebuilder:validation:Required
+	// +required
+	Policy Policy `json:"policy"`
+}
+
+// ImageScope is the item of the scopes list.
+// +kubebuilder:validation:Pattern=`^\*(?:\.(?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9]))+$|^((((?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9])(?:\.(?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9]))+(?::[0-9]+)?)|(localhost(?::[0-9]+)?))(?:(?:/[a-z0-9]+(?:(?:(?:[._]|__|[-]*)[a-z0-9]+)+)?)+)?)(?::([\w][\w.-]{0,127}))?(?:@([A-Za-z][A-Za-z0-9]*(?:[-_+.][A-Za-z][A-Za-z0-9]*)*[:][[:xdigit:]]{32,}))?$`
+type ImageScope string
+
+// Policy defines the verification policy for the items in the scopes list.
+type Policy struct {
+	// rootOfTrust specifies the public key, the Fulcio certificate and the Rekor public key.
```

Comment:
If I'm an end user, who wants to create a policy, is it expected that I know what Fulcio and Rekor are?

As an EP reviewer, these haven't been mentioned up to this point, what are they, can I learn more about them somewhere?

Reply:
I assume the user already knows the kind of certificates that are used to sign the image when they verify the signature. 

Fulcio and Rekor are under the sigstore project. https://github.com/sigstore/fulcio, https://github.com/sigstore/rekor

Reply:
I would probably not assume that, looking at the two links, they appear to basically be a key generator and a CA? It sounds like it's possible that you don't actually need to use Fulcio and Rekor to configure this feature? Is that right?

If those projects are not mandatory, it would probably be better to use more generic and well known terms than referring directly to the specific projects

Reply:
Ok, I think that makes sense as an argument for keeping these named then. I think there's a level of context we _should_ add to the godoc, but you're right, not a complete tutorial. We can link out to docs for Rekor and Fulcio though to provide an end user somewhere to learn more

A good explanation in this enhancement on how it works would certainly help the technical writer transforming this into product docs down the line